CreateAccount/index.js

```diff
@@ -38,66 +44,46 @@ module.exports = async function (context, req) {
         return;
     }
 
+    context.log.info("Request body validated. InvocationId: " + context.invocationId);
+
     try {
 
-        if (userAccountReq.role == "Administrator") {
-            const existingUser = await UserAccountModel.findOne({ $or: [{ email: userAccountReq.email, role: "Administrator" }, { username: userAccountReq.username, role: "Administrator" }] });
-            context.log("[DB FIND] - Find Existing User: ", existingUser);
-
-            if (existingUser) {
-                context.res = {
-                    status: 400,
-                    body: responseUtils.createResponse(true, false, errorMessages.USERNAME_EMAIL_ALREADY_USED, null)
-                }
-                context.done();
-                return;
-            }
-        }
+        const mongoClient = await mongoose.connect(process.env.MONGO_CONNECTION, context);
+
+        const service = new accountService(new mongooseRepository(mongoClient.model('UserAccount')), context);
```

Comment:
This makes sense to me. I would only instantiate and assign the repository to a variable before passing it to the constructor in order to improve readability

Reply:
Agree.